refractored_generator.rb

```diff
@@ -0,0 +1,70 @@
+# This creates a method that looks for a vowel. 
+def is_vowel?(letter)
+  case letter
+  when "a", "e", "i", "o", "u"
+      return true
+  else 
+      return false
+  end 
+end
+# Method gets user input for first and second word. Asks user to input again if enter does not meet parametes(must be at least 2 characters long.)
+def run_generator 
+    puts "What is the first word to combine?"
+    word1 = gets.chomp
+    until word1.length > 1 #&& word1 == "a" || word1 == "e" || word1 == "i" || word1 == "o" || word1 == "u"
+        puts "Input must be at least two characters long and include a vowel."
+        puts "Please enter the first word."
+        word1 = gets.chomp
+    
+    end 
+    puts "What is the second word to combine?"
+    word2 = gets.chomp
+    until word2.length > 1
+        puts "Input must be at least two characters long and include a vowel."
+        puts "Please enter the second word."
+        word2 = gets.chomp
+    end 
+ 
+#check if first word contains a vowel then deletes at the last vowel. 
+counter1 = 0
+rev_word1 = word1.reverse
+
+rev_word1.each_char do |letter|
+    if !is_vowel?(letter)
+        counter1 +=1
+    else
+        break
+    end
+end 
+
+puts "Word 1s first vowel when revered is at index #{counter1}"
+word_part1 = word1[0...((word1.length - 1)) - counter1]
+puts "The first half of your portmanteau is #{word_part1}"
+
+# In the second word, find the first instance of a vowel. Keep every letter of the second word 
+# starting after and including the first vowel. 
+counter2 = 0
```

Comment:
These steps make sense broken into separate methods.  

Notice how your `run_generator` method seems to run forever.  That's a good clue that it needs to be broken into sub methods.